src/app_service/service/token/service_main.nim

```diff
@@ -200,6 +200,15 @@ proc tokenAvailableForBridgingViaHop*(self: Service, tokenChainId: int, tokenAdd
   self.tokensForBridgingViaHop[key] = available
   return available
 
+## Returns the list of token keys available for swap via Paraswap for the given chainId, if the chain is not supported by Paraswap, the list will be empty.
+## NOTE: for now we don't store the list of tokens available for swap via Paraswap, we fetch it on demand.
+## Reason: storing can speed up switching the chain, but will increase the occupied memory.
+proc getListOfTokenKeysAvailableForSwapViaParaswap*(self: Service, chainId: int): seq[string] =
+  if chainId <= 0:
+    warn "invalid chainId", chainId = chainId
+    return @[]
+  return getListOfTokenKeysAvailableForSwapViaParaswap(chainId)
```

Comment:
Storing this response when the user runs the Swap flow for selected chain for the first time (or storing on the app start for all chains maybe? or when enable/disable chain?) should speed up the flow, since it has (~5261 tokens for ethereum chain), but will increase the memory. For now I left it as it is, fetching tokens on each Swap flow run.